{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  An Introduction To Applied Bioinformatics <a class='iab-edit' href='https://github.com/caporaso-lab/An-Introduction-to-Applied-Bioinformatics/edit/master/book/index.md#L2' target='_blank'>[edit]</a>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of Contents**\n",
    "0. [Getting started](1/index.ipynb)\n",
    "    0. [Reading An Introduction to Applied Bioinformatics](1/1.ipynb)\n",
    "        0. [Who should read IAB?](1/1.ipynb#1)\n",
    "        0. [How to read IAB](1/1.ipynb#2)\n",
    "        0. [Using the IPython Notebook](1/1.ipynb#3)\n",
    "        0. [Reading list](1/1.ipynb#4)\n",
    "            0. [Getting started with Biology](1/1.ipynb#4.1)\n",
    "            0. [Getting started with Computer Science and programming](1/1.ipynb#4.2)\n",
    "            0. [Philosophy of biology and popular science books](1/1.ipynb#4.3)\n",
    "        0. [Need help?](1/1.ipynb#5)\n",
    "        0. [Contributing to IAB](1/1.ipynb#6)\n",
    "        0. [About the author](1/1.ipynb#7)\n",
    "        0. [Acknowledgements](1/1.ipynb#8)\n",
    "0. [Fundamentals](2/index.ipynb)\n",
    "    0. [Pairwise sequence alignment](2/1.ipynb)\n",
    "        0. [What is a sequence alignment?](2/1.ipynb#1)\n",
    "        0. [A simple procedure for aligning a pair of sequences](2/1.ipynb#2)\n",
    "            0. [Step 1: Create a blank matrix where the rows and columns represent the positions in the sequences.](2/1.ipynb#2.1)\n",
    "            0. [Step 2: Add values to the cells in the matrix.](2/1.ipynb#2.2)\n",
    "            0. [Step 3: Identify the longest diagonals.](2/1.ipynb#2.3)\n",
    "            0. [Step 4: Transcribe some of the possible alignments that arise from this process.](2/1.ipynb#2.4)\n",
    "            0. [Why this simple procedure is too simplistic](2/1.ipynb#2.5)\n",
    "        0. [Differential scoring of matches and mismatches](2/1.ipynb#3)\n",
    "        0. [A better approach for global pairwise alignment using the Needleman-Wunsch algorithm](2/1.ipynb#4)\n",
    "            0. [Stepwise Needleman-Wunsch alignment](2/1.ipynb#4.1)\n",
    "                0. [Step 1: Create blank matrices.](2/1.ipynb#4.1.1)\n",
    "                0. [Step 2: Compute $F$ and $T$.](2/1.ipynb#4.1.2)\n",
    "                0. [Step 3: Transcribe the alignment.](2/1.ipynb#4.1.3)\n",
    "            0. [Automating Needleman-Wunsch alignment with Python](2/1.ipynb#4.2)\n",
    "            0. [A note on computing $F$ and $T$](2/1.ipynb#4.3)\n",
    "        0. [Global versus local alignment](2/1.ipynb#5)\n",
    "        0. [Smith-Waterman local sequence alignment](2/1.ipynb#6)\n",
    "            0. [Step 1: Create blank matrices.](2/1.ipynb#6.1)\n",
    "            0. [Step 2: Compute $F$ and $T$.](2/1.ipynb#6.2)\n",
    "            0. [Step 3: Transcribe the alignment.](2/1.ipynb#6.3)\n",
    "            0. [Automating Smith-Waterman alignment with Python](2/1.ipynb#6.4)\n",
    "        0. [Differential scoring of gaps](2/1.ipynb#7)\n",
    "        0. [How long does pairwise sequence alignment take?](2/1.ipynb#8)\n",
    "            0. [Comparing implementations of Smith-Waterman](2/1.ipynb#8.1)\n",
    "            0. [Analyzing Smith-Waterman run time as a function of sequence length](2/1.ipynb#8.2)\n",
    "            0. [Conclusions on the scalability of pairwise sequence alignment with Smith-Waterman](2/1.ipynb#8.3)\n",
    "    0. [Sequence homology searching](2/2.ipynb)\n",
    "        0. [Defining the problem](2/2.ipynb#1)\n",
    "        0. [Loading annotated sequences](2/2.ipynb#2)\n",
    "        0. [Defining the problem](2/2.ipynb#3)\n",
    "        0. [A complete homology search function](2/2.ipynb#4)\n",
    "        0. [Reducing the runtime for database searches](2/2.ipynb#5)\n",
    "        0. [Heuristic algorithms](2/2.ipynb#6)\n",
    "            0. [Random reference sequence selection](2/2.ipynb#6.1)\n",
    "            0. [Composition-based reference sequence collection](2/2.ipynb#6.2)\n",
    "                0. [GC content](2/2.ipynb#6.2.1)\n",
    "                0. [kmer content](2/2.ipynb#6.2.2)\n",
    "                0. [Further optimizing composition-based approaches by pre-computing reference database information](2/2.ipynb#6.2.3)\n",
    "        0. [Determining the statistical significance of a pairwise alignment](2/2.ipynb#7)\n",
    "            0. [Metrics of alignment quality](2/2.ipynb#7.1)\n",
    "            0. [False positives, false negatives, p-values, and alpha](2/2.ipynb#7.2)\n",
    "            0. [Interpreting alignment scores in context](2/2.ipynb#7.3)\n",
    "            0. [Exploring the limit of detection of sequence homology searches](2/2.ipynb#7.4)\n",
    "    0. [Generalized dynamic programming for multiple sequence alignment](2/3.ipynb)\n",
    "        0. [Progressive alignment](2/3.ipynb#1)\n",
    "            0. [Building the guide tree](2/3.ipynb#1.1)\n",
    "            0. [Generalization of Needleman-Wunsch (with affine gap scoring) for progressive multiple sequence alignment](2/3.ipynb#1.2)\n",
    "            0. [Putting it all together: progressive multiple sequence alignment](2/3.ipynb#1.3)\n",
    "        0. [Progressive alignment versus iterative alignment](2/3.ipynb#2)\n",
    "    0. [Phylogenetic reconstruction](2/4.ipynb)\n",
    "        0. [Why build phylogenies?](2/4.ipynb#1)\n",
    "        0. [How phylogenies are reconstructed](2/4.ipynb#2)\n",
    "        0. [Some terminology](2/4.ipynb#3)\n",
    "        0. [Simulating evolution](2/4.ipynb#4)\n",
    "            0. [A cautionary word about simulations](2/4.ipynb#4.1)\n",
    "        0. [Visualizing trees with ete3](2/4.ipynb#5)\n",
    "        0. [Distance-based approaches to phylogenetic reconstruction](2/4.ipynb#6)\n",
    "            0. [Distances and distance matrices](2/4.ipynb#6.1)\n",
    "            0. [Alignment-free distances between sequences](2/4.ipynb#6.2)\n",
    "            0. [Alignment-based distances between sequences](2/4.ipynb#6.3)\n",
    "            0. [Jukes-Cantor correction of observed distances between sequences](2/4.ipynb#6.4)\n",
    "            0. [Phylogenetic reconstruction with UPGMA](2/4.ipynb#6.5)\n",
    "                0. [Applying UPGMA from SciPy](2/4.ipynb#6.5.1)\n",
    "                0. [Understanding the name](2/4.ipynb#6.5.2)\n",
    "            0. [Phylogenetic reconstruction with neighbor-joining](2/4.ipynb#6.6)\n",
    "            0. [Limitations of distance-based approaches](2/4.ipynb#6.7)\n",
    "        0. [Bootstrap analysis](2/4.ipynb#7)\n",
    "        0. [Parsimony-based approaches to phylogenetic reconstruction](2/4.ipynb#8)\n",
    "            0. [How many possible phylogenies are there for a given collection of sequences?](2/4.ipynb#8.1)\n",
    "        0. [Statistical approaches to phylogenetic reconstruction](2/4.ipynb#9)\n",
    "            0. [Bayesian methods](2/4.ipynb#9.1)\n",
    "            0. [Maximum likelihood methods](2/4.ipynb#9.2)\n",
    "        0. [Rooted versus unrooted trees](2/4.ipynb#10)\n",
    "        0. [Acknowledgements](2/4.ipynb#11)\n",
    "    0. [Sequence mapping and clustering](2/5.ipynb)\n",
    "        0. [De novo clustering of sequences by similarity](2/5.ipynb#1)\n",
    "            0. [Furthest neighbor clustering](2/5.ipynb#1.1)\n",
    "            0. [Nearest neighbor clustering](2/5.ipynb#1.2)\n",
    "            0. [Centroid clustering](2/5.ipynb#1.3)\n",
    "            0. [Three different definitions of OTUs](2/5.ipynb#1.4)\n",
    "        0. [Comparing properties of our clustering algorithms](2/5.ipynb#2)\n",
    "        0. [Reference-based clustering to assist with parallelization](2/5.ipynb#3)\n",
    "0. [Applications](3/index.ipynb)\n",
    "    0. [Studying Microbial Diversity](3/1.ipynb)\n",
    "        0. [Getting started: the feature table](3/1.ipynb#1)\n",
    "        0. [Terminology](3/1.ipynb#2)\n",
    "        0. [Measuring alpha diversity](3/1.ipynb#3)\n",
    "            0. [Observed species (or Observed OTUs)](3/1.ipynb#3.1)\n",
    "                0. [A limitation of OTU counting](3/1.ipynb#3.1.1)\n",
    "            0. [Phylogenetic Diversity (PD)](3/1.ipynb#3.2)\n",
    "            0. [Even sampling](3/1.ipynb#3.3)\n",
    "        0. [Measuring beta diversity](3/1.ipynb#4)\n",
    "            0. [Distance metrics](3/1.ipynb#4.1)\n",
    "                0. [Bray-Curtis](3/1.ipynb#4.1.1)\n",
    "                0. [Unweighted UniFrac](3/1.ipynb#4.1.2)\n",
    "                0. [Even sampling](3/1.ipynb#4.1.3)\n",
    "            0. [Interpreting distance matrices](3/1.ipynb#4.2)\n",
    "                0. [Distribution plots and comparisons](3/1.ipynb#4.2.1)\n",
    "                0. [Hierarchical clustering](3/1.ipynb#4.2.2)\n",
    "            0. [Ordination](3/1.ipynb#4.3)\n",
    "                0. [Polar ordination](3/1.ipynb#4.3.1)\n",
    "                0. [Determining the most important axes in polar ordination](3/1.ipynb#4.3.2)\n",
    "                0. [Interpreting ordination plots](3/1.ipynb#4.3.3)\n",
    "        0. [Tools for using ordination in practice: scikit-bio, pandas, and matplotlib](3/1.ipynb#5)\n",
    "        0. [PCoA versus PCA: what's the difference?](3/1.ipynb#6)\n",
    "        0. [Are two different analysis approaches giving me the same result?](3/1.ipynb#7)\n",
    "            0. [Procrustes analysis](3/1.ipynb#7.1)\n",
    "        0. [Where to go from here](3/1.ipynb#8)\n",
    "        0. [Acknowledgements](3/1.ipynb#9)\n",
    "0. [Exercises](4/index.ipynb)\n",
    "    0. [Local sequence alignment exercises](4/1.ipynb)\n",
    "        0. [Purpose](4/1.ipynb#1)\n",
    "        0. [Background](4/1.ipynb#2)\n",
    "        0. [Goals](4/1.ipynb#3)\n",
    "        0. [Hints](4/1.ipynb#4)\n",
    "        0. [Getting started](4/1.ipynb#5)\n",
    "        0. [Question 1](4/1.ipynb#6)\n",
    "        0. [Question 2](4/1.ipynb#7)\n",
    "        0. [Question 3](4/1.ipynb#8)\n",
    "        0. [Question 4](4/1.ipynb#9)\n",
    "        0. [More hints](4/1.ipynb#10)\n",
    "    0. [Multiple sequence alignment exercises](4/2.ipynb)\n",
    "        0. [Purpose](4/2.ipynb#1)\n",
    "        0. [Goals](4/2.ipynb#2)\n",
    "        0. [Hints](4/2.ipynb#3)\n",
    "        0. [Functions that you will need to complete the exercise.](4/2.ipynb#4)\n",
    "        0. [Question 1](4/2.ipynb#5)\n",
    "        0. [Question 2](4/2.ipynb#6)\n",
    "        0. [Question 3](4/2.ipynb#7)\n",
    "        0. [Question 4](4/2.ipynb#8)\n",
    "        0. [Question 5](4/2.ipynb#9)\n",
    "        0. [Question 6](4/2.ipynb#10)\n",
    "        0. [Question 7](4/2.ipynb#11)\n",
    "0. [Back Matter](5/index.ipynb)\n",
    "    0. [Glossary](5/1.ipynb)\n",
    "        0. [Pairwise alignment (noun)](5/1.ipynb#1)"
   ]
  }
 ],
 "metadata": {},
 "nbformat": 4,
 "nbformat_minor": 2
}
